test/e2e/create_test.go

```diff
@@ -40,6 +42,78 @@ func creationTestSuite(t *testing.T) {
 	t.Run("Network validation", testNetwork)
 	t.Run("Label validation", func(t *testing.T) { testWorkerLabel(t) })
 	t.Run("NodeTaint validation", func(t *testing.T) { testNodeTaint(t) })
+	t.Run("machineCreation", func(t *testing.T) { testWindowsMachineCreation(t) })
+}
+
+// testWindowsMachineCreation tests the creation of the Windows machines and checks if WMCO is properly watching
+// for the Windows machines created
+func testWindowsMachineCreation(t *testing.T) {
+	testCtx, err := NewTestContext(t)
+	require.NoError(t, err)
+	defer testCtx.cleanup()
+	testCases := []struct {
+		testCase  string
+		isWindows bool
+	}{
+		{
+			testCase:  "Create a Windows VM with label",
+			isWindows: true,
+		},
+		{
+			testCase:  "Create a Windows VM without label",
+			isWindows: false,
+		},
+	}
+	var machineSetNames = make([]string, 0, len(testCases))
+	expectedEventCount := 0
+	for _, test := range testCases {
+		machineSet, err := testCtx.CloudProvider.GenerateMachineSet(test.isWindows)
+		require.NoError(t, err)
+		err = framework.Global.Client.Create(context.TODO(), machineSet,
+			&framework.CleanupOptions{TestContext: testCtx.osdkTestCtx, Timeout: cleanupTimeout,
+				RetryInterval: cleanupRetryInterval})
+		require.NoError(t, err)
+		log.Printf("MachineSet Name %s", machineSet.Name)
+		machineSetNames = append(machineSetNames, machineSet.Name)
+		if test.isWindows {
+			expectedEventCount++
+		}
+	}
+	require.NoError(t, testCtx.waitForProvisionedEvent(expectedEventCount, machineSetNames))
+}
+
+// waitForProvisionedEvent waits for 2 minutes for the required number of machines to come to the provisioned state.
+func (tc *testContext) waitForProvisionedEvent(expectedEventCount int, machineSetNames []string) error {
+	var actualEventCount int
+	timeOut := 2 * time.Minute
+	startTime := time.Now()
+	for i := 0; ; i++ {
+		currTime := time.Since(startTime)
+		if currTime >= timeOut {
+			break
+		}
```

Comment:
Can't this be done within the `for`?
```
startTime := time.Now()
timeOut := 2* time.Minute
for i := 0; time.Since(startTime) < timeOut; i++ {
```